### PR TITLE
ClusterLoader - Adding template functions

### PR DESCRIPTION
**Pull request by krzysied**
Adding generating random int functions.

### CODE REVIEW

File: clusterloader2/pkg/config/templatefunctions.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+func randInt(i int) int {
+	return rand.Intn(i)
+}
+
+func randIntRange(i, j int) int {
+	if i >= j {
+		return i
+	}
+	r := i + rand.Intn(j-i)
+	fmt.Printf("Rand: %v\n", r)
```
Comment (krzysied) [inline clusterloader2/pkg/config/templatefunctions.go]: My mistake - I'm going to remove this printf.

File: clusterloader2/pkg/config/templatefunctions.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+import (
+	"math/rand"
+	"text/template"
+)
+
+// GetFuncs returns map of names to functions, that are supported by template provider.
+func GetFuncs() template.FuncMap {
+	return template.FuncMap{
+		"randInt":      randInt,
```
Comment (shyamjvs) [inline clusterloader2/pkg/config/templatefunctions.go]: I'd suggest making these function names capitals (to show clearly they're constants).

Comment (krzysied) [reply clusterloader2/pkg/config/templatefunctions.go]: Done.

File: clusterloader2/pkg/config/templatefunctions.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+// GetFuncs returns map of names to functions, that are supported by template provider.
+func GetFuncs() template.FuncMap {
+	return template.FuncMap{
+		"RandInt":      randInt,
+		"RandIntRange": randIntRange,
+	}
+}
+
+func randInt(i int) int {
```
Comment (shyamjvs) [inline clusterloader2/pkg/config/templatefunctions.go]: Could you add comment above each of these 2 functions what's the range they create randoms from? (for e.g 0-based or 1-based, open or closed interval)

Comment (krzysied) [reply clusterloader2/pkg/config/templatefunctions.go]: Comments added.

File: clusterloader2/pkg/config/templatefunctions.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+		"RandIntRange": randIntRange,
+	}
+}
+
+// randInt returns pseudo-random int in [0, i).
+func randInt(i int) int {
+	return rand.Intn(i)
+}
+
+// randIntRange returns pseudo-random int in [i, j).
```
[the remark this answers is not in the document]
Comment (krzysied) [reply clusterloader2/pkg/config/templatefunctions.go]: Should I also change the `RandInt` function or only `RandIntRange`?
Btw, using [..) set is common programming practice - I assume that people with expect this here.

[the remark this answers is not in the document]
Comment (shyamjvs) [reply clusterloader2/pkg/config/templatefunctions.go]: I'd suggest changing both (for consistency).

> Btw, using [..) set is common programming practice - I assume that people with expect this here.

Yeah, that's a matter of convention (you're right that few pkgs assume open interval). But I think this'll be more convenient when we calculate and put the limits in the config.

[the remark this answers is not in the document]
Comment (krzysied) [reply clusterloader2/pkg/config/templatefunctions.go]: OK. Done.

File: clusterloader2/pkg/config/templatefunctions.go [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+func GetFuncs() template.FuncMap {
+	return template.FuncMap{
+		"RandInt":      randInt,
+		"RandIntRange": randIntRange,
+	}
+}
+
+// randInt returns pseudo-random int in [0, i].
+func randInt(i int) int {
+	return rand.Intn(i + 1)
```
Comment (shyamjvs) [inline clusterloader2/pkg/config/templatefunctions.go]: 'i + 1' has spaces but similar expression in below function doesn't. I thought gofmt should be failing on this.

**Pull request merged**